w09h02/test/pgdp/trains/processing/utils/JSONParser.java

```diff
@@ -0,0 +1,53 @@
+package pgdp.trains.processing.utils;
+
+import org.json.JSONObject;
+import pgdp.trains.connections.TrainConnection;
+import pgdp.trains.processing.resources.JSONDataset;
+
+public class JSONParser {
+  private static final JSONObject iceBasicDataset;
+  private static final TrainConnection iceBasicParsed;
+  private static final JSONObject str29CanceledDataset;
+  private static final TrainConnection str29CanceledParsed;
+  private static final JSONObject re8BasicDataset;
+  private static final TrainConnection re8BasicParsed;
+
+  static {
+    iceBasicDataset = new JSONObject(JSONDataset.iceBasicBlob);
+    str29CanceledDataset = new JSONObject(JSONDataset.str29CanceledBlob);
+    re8BasicDataset = new JSONObject(JSONDataset.re8BasicBlob);
+
+    iceBasicParsed = TrainConnection.parseFromJSON(iceBasicDataset);
```

Comment:
Could you replace the repetitive json parse method with the given static method?

```suggestion
    iceBasicParsed = getTrainConnectionFromJSON(iceBasicDataset);
```

Reply:
Then you would have another layer of indirection. I created the shell method specifically for converting when running unit tests.